### PR TITLE
Market: fix rest responses

### LINKED ISSUE #450: Fix unsubscribe's result type
Need fix in two repos
- [x] ya-client (as per https://github.com/golemfactory/ya-client/issues/21 addressed by https://github.com/golemfactory/ya-client/pull/59)
- [x] here (#765)

### PR DESCRIPTION
**Pull request by tworec**
It breaks compatibility of Market API, so **not to be merged before `Alpha.3`.**
fixes #450 
depends on: golemfactory/ya-client#59

Contains changes from #850 and #869

### CODE REVIEW

**Review by Wiezzel — DISMISSED** [review]
Look OK, but please merge the changes in `ya-client` to master and update commit hash in `Cargo.toml`.

File: core/market/src/rest_api/provider.rs [hunk 1]
```diff
@@ -153,7 +153,7 @@ async fn approve_agreement(
         .approve_agreement(id, &agreement_id, timeout)
         .await
         .log_err()
-        .map(|_| HttpResponse::NoContent().finish())
+        .map(|status| HttpResponse::Ok().json(status.to_string()))
```
Comment (tworec) [inline core/market/src/rest_api/provider.rs:156]: should be `410` *Gone* as suggested by @prekucki

Comment (tworec) [reply core/market/src/rest_api/provider.rs:155]: Addressed here https://github.com/golemfactory/ya-client/pull/70

**Review by jiivan — APPROVED** [review]
lgtm

File: Cargo.toml [hunk 2]
```diff
@@ -171,8 +171,11 @@ ya-sb-router = { path = "service-bus/router" }
 ya-sb-util = { path = "service-bus/util" }
 
 ## CLIENT
-ya-client = { git = "https://github.com/golemfactory/ya-client.git", rev = "80221a38c4cf8f4700da36598316fc8048e5c535"}
-ya-client-model = { git = "https://github.com/golemfactory/ya-client.git", rev = "80221a38c4cf8f4700da36598316fc8048e5c535"}
+ya-client = { git = "https://github.com/golemfactory/ya-client.git", rev = "ade8c91ef6cf4202eb2390c266f89fa0ec1f25e4"}
+ya-client-model = { git = "https://github.com/golemfactory/ya-client.git", rev = "ade8c91ef6cf4202eb2390c266f89fa0ec1f25e4"}
+
+#ya-client = { path = "../ya-client" }
+#ya-client-model = { path = "../ya-client/model" }
```
Comment (mfranciszkiewicz) [inline Cargo.toml:178]: Please remove

**Pull request merged**